src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -780,17 +783,47 @@ internal void Assign(IEnumerable<TopicPartitionOffset> partitions)
                 }
             }
 
-            ErrorCode err = Librdkafka.assign(handle, list);
+            ErrorCode err = ErrorCode.NoError;
+            Error error = null;
+
+            if (assignMethodErr != null) { err = assignMethodErr(handle, list); }
+            if (assignMethodError != null) { error = new Error(assignMethodError(handle, list)); }
+
             if (list != IntPtr.Zero)
             {
                 Librdkafka.topic_partition_list_destroy(list);
             }
-            if (err != ErrorCode.NoError)
+
+            if (err != ErrorCode.NoError) { throw new KafkaException(CreatePossiblyFatalError(err, null)); }
+            if (error != null && error.Code != ErrorCode.NoError) { throw new KafkaException(error); }
```

Comment:
else if

Reply:
oh, above comment was meant for here. made else anyway here too.